lib/rialto/etl/extractors/sera.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'faraday_middleware'
+require 'oauth2'
+
+module Rialto
+  module Etl
+    module Extractors
+      # Documetation: https://asconfluence.stanford.edu/confluence/display/MaIS/SeRA+API+-+User+Documentation
+      class Sera
+        def initialize(options = {})
+          @sunetid = options.fetch(:sunetid)
+        end
+
+        # Hit an API endpoint and return the results
+        def each(&block)
+          body.each(&block)
+        rescue StandardError => exception
+          warn "Error: #{exception.message}"
+        end
+
+        private
+
+        attr_reader :sunetid
+
+        def client
+          @client ||= Faraday.new ::Settings.sera.service_url do |conn|
+            conn.request :oauth2, token, token_type: :bearer
+            conn.adapter Faraday.default_adapter
+          end
+        end
+
+        def token
+          client = OAuth2::Client.new(::Settings.tokens.sera.clientid,
+                                      ::Settings.tokens.sera.secret,
+                                      token_url: ::Settings.sera.token_url)
+          client.client_credentials.get_token.token
+        end
+
+        # @return[Array<Hash>] the results of the API call
+        def body
+          case response.status
+          when 404
+            []
```

Comment:
What is the significance of a 404 in this context? I wonder if a comment would be useful here to explain why a 404 should cause an empty result set.